tests/data_context/store/test_expectations_store.py

```diff
@@ -310,8 +310,177 @@ def test_get_key_in_non_cloud_mode(empty_data_context):
+        column="a",
+        value_set=[1, 2, 3],
+        result_format="BASIC",
+    )
+
+    # Act
+    store.add_expectation(suite=suite, expectation=expectation)
+
+    # Assert
+    updated_suite_dict = store.get(key=key)
+    updated_suite = ExpectationSuite(**updated_suite_dict)
+    added_expectation = updated_suite.expectations[0]
+    assert UUID(added_expectation.id)
+    assert expectation.column == added_expectation.column
+    assert expectation.value_set == added_expectation.value_set
+    assert expectation.result_format == added_expectation.result_format
+
+
+@pytest.mark.filesystem
+def test_add_expectation_disregards_provided_id(empty_data_context):
+    # Arrange
+    context = empty_data_context
+    store = context.expectations_store
+    suite_name = "test-suite"
+    suite = context.add_expectation_suite(suite_name)
+    key = store.get_key(suite)
+    provided_id = "e86bb8a8-b75f-4efb-a3bb-210b6440661e"
+    expectation = ExpectColumnValuesToBeInSet(
+        id=provided_id,
+        column="a",
+        value_set=[1, 2, 3],
+        result_format="BASIC",
+    )
+
+    # Act
+    store.add_expectation(suite=suite, expectation=expectation)
+
+    # Assert
+    updated_suite_dict = store.get(key=key)
+    updated_suite = ExpectationSuite(**updated_suite_dict)
+    added_expectation = updated_suite.expectations[0]
+    assert UUID(added_expectation.id)
+    assert added_expectation.id != provided_id
+
+
+@pytest.mark.cloud
+def test_update_expectation_success_cloud_backend(empty_cloud_data_context):
+    # Arrange
+    context = empty_cloud_data_context
+    store = context.expectations_store
+    suite_name = "test-suite"
+    expectation = ExpectColumnValuesToBeInSet(
+        column="a",
+        value_set=[1, 2, 3],
+        result_format="BASIC",
+    )
+    suite = context.add_expectation_suite(
+        suite_name, expectations=[expectation.configuration]
+    )
+    key = store.get_key(suite)
+    expectation = suite.expectations[0]
+    updated_column_name = "foo"
+    expectation.column = updated_column_name
+
+    # Act
+    store.update_expectation(suite=suite, expectation=expectation)
+
+    # Assert
+    updated_suite_dict = store.get(key=key)
+    updated_suite = ExpectationSuite(**updated_suite_dict)
+    updated_expectation = updated_suite.expectations[0]
+    assert updated_expectation.id == expectation.id
+    assert updated_expectation.column == updated_column_name
+
+
+@pytest.mark.filesystem
+def test_update_expectation_success_file_backend(empty_data_context):
+    # Arrange
+    context = empty_data_context
+    store = context.expectations_store
+    suite_name = "test-suite"
+    expectation = ExpectColumnValuesToBeInSet(
+        column="a",
+        value_set=[1, 2, 3],
+        result_format="BASIC",
+    )
+    suite = context.add_expectation_suite(
+        suite_name, expectations=[expectation.configuration]
+    )
+    key = store.get_key(suite)
+    expectation = suite.expectations[0]
```

Comment:
I've move these lines to the act section. I think the arrange is setting up the expectation suite with fixed data. The action is retrieving the expectation and updating it.

Reply:
[b5f13be](https://github.com/great-expectations/great_expectations/pull/9125/commits/b5f13bea3246ec811d5b0f5e59e1f730bd4cb9a1)